event.js

```diff
@@ -15,7 +23,8 @@ module.exports = function awaitEvent({ eventEmitter, resolveOn = [], rejectOn =
             return function (...eventArguments)
             {
                 unregister();
-                reject(new aFunction(anEventName, eventArguments));
+                const error = typeof aFunction === "function" ? new aFunction(anEventName, eventArguments) : aFunction
```

Comment:
Is this just defensive programming? Does it happen that aFunction might not be a function?

Reply:
Since the Timeout error is created when the promise is create, it gets passed to `errorOut()` as the error object itself, not the constructor function. But for regular `EventError`s the constructor function is still passed like it was originally.

Reply:
Sure, makes sense to me.